dali/base/dasds.cpp

```diff
@@ -6361,46 +6312,112 @@ void CCovenSDSManager::checkEnvironment(IPropertyTree *oldEnv, IPropertyTree *ne
     }
 }
 
-void CCovenSDSManager::loadStore(const char *storeName, const bool *abort)
+CServerRemoteTree *CCovenSDSManager::loadStoreType(StoreFormat storeFormat, size32_t bufferSize, const bool *abort)
 {
-    if (root) root->Release();
+    StringBuffer storeFilename(dataPath);
+    unsigned storedCrc = 0;
+    iStoreHelper->getCurrentStoreFilename(storeFilename, storeFormat, &storedCrc);
 
-    class CNodeCreate : implements IPTreeNodeCreator, public CInterface
+    class CNodeCreate : public CSimpleInterfaceOf<IPTreeNodeCreator>
     {
     public:
-        IMPLEMENT_IINTERFACE;
         virtual IPropertyTree *create(const char *tag) { return createServerTree(tag); }
-    } nodeCreator;
-    class CSDSTreeMaker : public CPTreeMaker
+    };
+    Owned<IPTreeNodeCreator> nodeCreator = new CNodeCreate();
+
+    bool isBinary = (storeFormat == StoreFormat::BINARY);
+    try
     {
-    public:
-        CSDSTreeMaker(IPTreeNodeCreator *nodeCreator) : CPTreeMaker(ipt_none, nodeCreator) { }
-        virtual void endNode(const char *tag, unsigned length, const void *value, bool binary, offset_t endOffset)
+        Owned<IPropertyTree> result;
+
+        OwnedIFile iFileStore = createIFile(storeFilename);
+        OwnedIFileIO iFileIOStore = iFileStore->open(IFOread);
+        if (!iFileIOStore)
+            throw MakeSDSException(SDSExcpt_OpenStoreFailed, "%s", storeFilename.str());
+
+        offset_t fSize = iFileIOStore->size();
+        unsigned crc{0};
+        PROGLOG("Loading %s store %u (size=%.2f MB, storedCrc=%x)", isBinary ? "Binary" : "XML", iStoreHelper->queryCurrentEdition(), ((double)fSize) / 0x100000, storedCrc);
+
+        Owned<ISerialInputStream> serialStream = createSerialInputStream(iFileIOStore);
+        Owned<ISerialInputStream> progressStream = createProgressStream(serialStream, 0, fSize, "Load progress", 60);
+        Owned<ICrcSerialInputStream> crcSerialStream = createCrcInputStream(progressStream);
+        Owned<IBufferedSerialInputStream> bufInStream = createBufferedInputStream(crcSerialStream, bufferSize);
+        if (isBinary)
+            result.setown(createPTreeFromBinary(*bufInStream, nodeCreator));
+        else
         {
-            IPropertyTree *node = queryCurrentNode();
-            CPTreeMaker::endNode(tag, length, value, binary, endOffset);
-            if (((CServerRemoteTree *)node)->testExternalCandidate())
-                convertQueue.append(*(CServerRemoteTree *)node);
```

Comment:
NB: this mechanism has been deliberately deprecated as part of this PR.
It used to hook the XML parser, and check every value against the currently configured external threshold. If bigger, it added to this convertQueue list, and then converted them to externals.
It did not check if values were now smaller than the threshold and convert them the other way around.

The threshold hasn't changed in over a decade, and it was difficult to support in the binary deserialization version, so removed.